### PR TITLE
Bump PHPStan level to 5

### PR DESCRIPTION
**Pull request by phansys**
**TODO**:
- [x] Wait for #2603.

### CODE REVIEW

**Review by franmomu — COMMENTED** [review]
This round of comments, I'll try to continue later

File: src/Mapping/Annotation/Uploadable.php [hunk 1]
```diff
@@ -54,6 +55,8 @@ final class Uploadable implements GedmoAnnotation
 
     /**
      * @var string
+     *
+     * @phpstan-var Validator::FILENAME_GENERATOR_SHA1|Validator::FILENAME_GENERATOR_ALPHANUMERIC|Validator::FILENAME_GENERATOR_NONE|class-string<FilenameGeneratorInterface>
```
[the remark this answers is not in the document]
Comment (phansys) [reply src/Mapping/Annotation/Uploadable.php:59]: You're right. I didn't take this shortcut into account. Thank you.

File: src/Mapping/MappedEventSubscriber.php [hunk 2]
```diff
@@ -103,9 +109,29 @@ public function __construct()
...
+     *  maxSize?: float,
+     *  path?: string,
+     *  pathMethod?: string,
+     *  allowedTypes?: string[],
+     *  disallowedTypes?: string[],
+     *  filenameGenerator?: MappingValidator::FILENAME_GENERATOR_ALPHANUMERIC|MappingValidator::FILENAME_GENERATOR_SHA1|MappingValidator::FILENAME_GENERATOR_NONE|class-string<FilenameGeneratorInterface>,
+     *  fileMimeTypeField?: string,
+     *  fileSizeField?: string,
+     *  callback?: string,
+     * }
```
[the remark this answers is not in the document]
Comment (phansys) [reply src/Mapping/MappedEventSubscriber.php:134]: This is the base class for the listeners of all the extensions. By the moment, the options listed here are the result of all the options reported by PHPStan. I aggree that it should be splitted by extension and moved to each specific extension using `@phpstan-method`, but I'd like to left this task for a future round.

File: src/DoctrineExtensions.php [hunk 3]
```diff
@@ -122,7 +123,9 @@ private static function createAnnotationReader(): AnnotationReader
         if (class_exists(ArrayAdapter::class)) {
             $reader = new PsrCachedReader($reader, new ArrayAdapter());
         } elseif (class_exists(ArrayCache::class)) {
-            $reader = new PsrCachedReader($reader, CacheAdapter::wrap(new ArrayCache()));
+            $cache = new ArrayCache();
+            assert($cache instanceof Cache);
+            $reader = new PsrCachedReader($reader, CacheAdapter::wrap($cache));
         }
```
[the remark this answers is not in the document]
Comment (phansys) [reply src/DoctrineExtensions.php:129]: Good point. Let me verify this.

[the remark this answers is not in the document]
Comment (phansys) [reply src/DoctrineExtensions.php]: Done!

**Review by franmomu — APPROVED** [review]

**Comment by franmomu** [comment]
thanks @phansys!

**Pull request merged**